cmd/syncEnumerator.go

```diff
@@ -34,10 +34,30 @@ import (
 
 func (cca *cookedSyncCmdArgs) initEnumerator(ctx context.Context) (enumerator *syncEnumerator, err error) {
 
+	srcCredInfo, srcIsPublic, err := getCredentialInfoForLocation(ctx, cca.fromTo.From(), cca.source.Value, cca.source.SAS, true)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if cca.fromTo.IsS2S() {
+		if cca.fromTo.From() == common.ELocation.Blob() || cca.fromTo.From() == common.ELocation.File() {
```

Comment:
What's the purpose of this test?  Is it to exclude the S3 scenario?  If so, why not write is as != s3?  That way, if/when we add ADLS Gen 2 sync one day, this code will still be correct for ADLS Gen 2 (instead of just getting skipped when we add ADLS Gen 2).

Reply:
That's probably a good point. I'll adjust it.